components/channel/OnReceivingData.js

```diff
@@ -9,31 +9,42 @@ import { isBinaryPayload, pascalCase, isStringPayload } from '../../utils/index'
  */
 export function OnReceivingData(message, defaultContentType) {
   //Check if we are converting from binary
-  let convertToBinary = '';
+  let convertFromBinary = '';
```

Comment:
I'm going to propose a batch of changes from this line and below:

```suggestion
  let convertFromBinary;
```